controllers/configuration_controller.go

```diff
@@ -89,16 +95,33 @@ func (r *ConfigurationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 	klog.InfoS("reconciling Terraform Configuration...", "NamespacedName", req.NamespacedName)
 	applyJobName := configurationName + "-" + string(TerraformApply)
 
-	// Terraform destroy
 	if err := r.Get(ctx, req.NamespacedName, &configuration); err != nil {
-		klog.InfoS("performing Terraform Destroy", "Namespace", req.Namespace, "Name", req.Name)
 		if kerrors.IsNotFound(err) {
+			klog.ErrorS(err, "unable to fetch Configuration", "NamespacedName", req.NamespacedName)
+			err = nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	if configuration.ObjectMeta.DeletionTimestamp.IsZero() {
+		if !controllerutil.ContainsFinalizer(&configuration, configurationFinalizer) {
+			controllerutil.AddFinalizer(&configuration, configurationFinalizer)
+			return ctrl.Result{}, errors.Wrap(r.Client.Update(ctx, &configuration), configurationFinalizer)
```

Comment:
There will be one error log in the controller pod, right?
There are many error logs in order to make the controller reconcile in the last logics.